lib/improver/tests/utilities/test_save.py

```diff
@@ -144,10 +156,72 @@ def test_cf_shared_attributes_list(self):
         global_keys = Dataset(self.filepath, mode='r').ncattrs()
-        self.assertEqual(len(global_keys), 1)
+        self.assertEqual(len(global_keys), 8)
         self.assertTrue(all(key in self.global_keys_ref
                             for key in global_keys))
 
 
+class Test_append_metadata_cube(IrisTest):
+    """Test that appropriate metadata cube and attributes have been appended
+            to the cubes in the cube list"""
+
+    def setUp(self):
+        """ Set up cube to write, read and check """
+        self.global_keys_ref = ['title', 'um_version', 'grid_id', 'source',
+                                'Conventions', 'institution', 'history',
+                                'bald__isPrefixedBy']
+        self.directory = mkdtemp()
+        self.filepath = os.path.join(self.directory, "temp.nc")
+        self.cube = set_up_test_cube()
+
+    def tearDown(self):
+        """ Remove temporary directories created for testing. """
+        call(['rm', '-f', self.filepath])
+        call(['rmdir', self.directory])
+
+    def test_bald_attribute_added(self):
+        """Test that the bald__isPrefixedBy attribute is added to each cube
+        and points to prefix_list"""
+        cube_list = ([self.cube, self.cube])
+        metadata_cubelist = append_metadata_cube(
+            cube_list, self.global_keys_ref)
+        for cube in metadata_cubelist:
+            self.assertTrue(
+                cube.attributes['bald__isPrefixedBy']
+                is 'prefix_list')
+
+    def test_prefix_cube_attributes(self):
+        """Test that metadata prefix cube contains the correct attributes"""
+        prefix_dict = {
+            'spp__': 'http://reference.metoffice.gov.uk/statistical-process'
+                     '/properties/',
+            'bald__isPrefixedBy': 'prefix_list',
+            'bald__': 'http://binary-array-ld.net/latest/',
+            'spv__': 'http://reference.metoffice.gov.uk/statistical-process'
+                     '/values/',
+            'rdf__': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+            'spd__': 'http://reference.metoffice.gov.uk/statistical-process'
+                     '/def/'}
+        metadata_cubelist = append_metadata_cube([], self.global_keys_ref)
+        self.assertDictEqual(metadata_cubelist[0].attributes, prefix_dict)
+
+    def test_global_attributes_present(self):
+        """Test that desired global attributes are added to the prefix cube
+        so that Iris2 keeps these attributes global in any resultant
+        netCDF file saved using these cubes"""
+
+        cube_list = ([self.cube, self.cube])
+
+        metadata_cubelist = append_metadata_cube(
+            cube_list, self.global_keys_ref)
+
+        keys_in_prefix_cube = metadata_cubelist[2].attributes
+        prefix_global_keys = [
+            k for k in keys_in_prefix_cube.keys() if k in self.global_keys_ref]
+
+        self.assertListEqual(
+            sorted(self.global_keys_ref), sorted(prefix_global_keys))
```

Comment:
Not sure this is comparing strictly the right thing.  We want to make sure that all global keys that were on the input cube are also now on the prefix cube.  Even though all of `self.global_keys_ref` are currently on `self.cube`, this is not by design and could lead to misinterpretation in the future.  Suggest the following:
```
prefix_global_keys = [
    k for k in keys_in_prefix_cube.keys() if k in self.global_keys_ref]
data_cube_global_keys = [
    k for k in self.cube.attributes.keys() if k in self.global_keys_ref]

self.assertListEqual(
    sorted(prefix_global_keys), sorted(data_cube_global_keys))
```

Reply:
Actually what might be even better is:
```
prefix_global_attrs = [
    attr for attr in keys_in_prefix_cube if attr.keys() in self.global_keys_ref]
data_global_attrs = [
    attr for attr in self.cube.attributes if attr.keys() in self.global_keys_ref]

self.assertDictEqual(prefix_global_attrs, data_global_attrs)
```
As I believe this will check both the attributes' keys and their values.

Reply:
We've established the above doesn't work.  It would still be good to check values using eg the following:
```
for key in prefix_global_keys:
    self.assertEqual(metadata_cubelist[-1].attributes[key], self.cube.attributes[keys])
```